File: sparkscala/scalademo/src/main/scala/com/neohope/spark/demos/scala/BasicSum.scala
package com.neohope.spark.demos.scala

import org.apache.spark._

/**
 * 求和
 */
object BasicSum {
  val sc = new SparkContext("local", "BasicMap")

  def computeSum(nums: List[Int]) = {
    val input = sc.parallelize(nums)
    val sum = input.fold(0)((x, y) => (x + y))
    sum
  }

  def main(args: Array[String]) {
    val nums = List(0, 1, 2, 3, 4, 5, 6, 7, 8, 9)
    val sum = computeSum(nums)
    println(sum)
  }
}

File: sparkjava/javademo/src/main/java/com/neohope/spark/demo/BasicSum.java
package com.neohope.spark.demo;

import java.util.Arrays;
import java.util.List;

import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;

/**
 * 平均值计算
 */
public class BasicSum {
	JavaSparkContext sc = new JavaSparkContext("local", "basicsum");

	public Integer calcSum(List<Integer> nums) {
		JavaRDD<Integer> rdd = sc.parallelize(nums);
		Integer result = rdd.fold(0, new BasicSumFunctions.Sum());
		return result;
	}

	public static void main(String[] args) throws Exception {
		Integer[] nums = { 0, 1, 2, 3, 4, 5, 6, 7, 8, 9 };
		BasicSum basicSum = new BasicSum();
		Integer sum = basicSum.calcSum(Arrays.asList(nums));
		System.out.println(sum);
	}
}
